base/number.jl

```diff
@@ -63,4 +63,6 @@ zero{T<:Number}(::Type{T}) = convert(T,0)
 one(x::Number)  = oftype(x,1)
 one{T<:Number}(::Type{T}) = convert(T,1)
 
+_default_type(::Type{Number}) = Int
```

Comment:
If you have `Base._promote_op(+, Int, Number)` that will return `Any`, so what `promote_op` currently does is to call `Base._promote_op(+, Int, _default_type(Number))` which with this would return `Int` and then it returns `typejoin(Int, Number, Int) == Number` (the types of the inputs and the sample return type).

For `Number` it seems that we could let the container as `Number`, but for other containers that might not be the best option. If you do `broadcast(*, 1.5, Integer[1])`, `Integer` as the final container wouldn't be right , so `promote_op` does `Base._promote_op(*, Float64, Int)` (that is `Float64`) and then returns `Real` (by `typejoin`ing these).